### PR TITLE
PIM-9091: Merge 3.2 into 4.0

### PR DESCRIPTION
**Pull request by mmetayer**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: tests/legacy/features/Behat/Decorator/TabElement/ComparisonPanelDecorator.php [hunk 1]
```diff
@@ -85,7 +85,11 @@ public function switchSource($source)
             return $toggle;
         }, 'Dropdown action menu was not found');
 
-        $toggle->click();
+        $this->spin(function() use ($toggle) {
```
Comment (mmetayer) [inline tests/legacy/features/Behat/Decorator/TabElement/ComparisonPanelDecorator.php]: Sometimes the toggle is found really quick, but is not clickable yet (the page is not fully rendered yet and there's another element above the toggle)

Comment (mmetayer) [reply tests/legacy/features/Behat/Decorator/TabElement/ComparisonPanelDecorator.php]: (should fix https://github.com/akeneo/pim-community-dev/issues/11591)

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Resources/translations/jsmessages.ca_ES.yml [hunk 2]
```diff
@@ -143,6 +143,7 @@ pim_import_export:
         import: Importació
         export: Exportació
         mass_edit: Edició massiva
+        mass_edit_rule: Configuració de les regles
```
Comment (pierallard) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Resources/translations/jsmessages.ca_ES.yml]: :notes: Despacito :notes:

File: src/Akeneo/Tool/Bundle/VersioningBundle/Event/PrePurgeVersionEvent.php [hunk 3]
```diff
@@ -6,6 +6,10 @@
 use Symfony\Component\EventDispatcher\Event;
 
 /**
+ * @deprecated Will be removed in 4.0
```
Comment (pierallard) [inline src/Akeneo/Tool/Bundle/VersioningBundle/Event/PrePurgeVersionEvent.php]: 4.0 or master? If 4.0... Remove them (there are several)

File: src/Akeneo/Tool/Bundle/VersioningBundle/Repository/VersionRepositoryInterface.php [hunk 4]
```diff
@@ -84,11 +84,13 @@ public function getPendingVersionsCount();
     public function findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null);
 
     /**
+     * @deprecated Will be removed in 4.0
```
Comment (pierallard) [inline src/Akeneo/Tool/Bundle/VersioningBundle/Repository/VersionRepositoryInterface.php]: Same

File: src/Akeneo/UserManagement/Bundle/Resources/translations/messages.fi_FI.yml [hunk 5]
```diff
@@ -40,6 +40,13 @@ pim_user:
       has_group: Ryhmässä
       active: Aktiivinen
       inactive: Ei aktiivinen
+    fields_errors:
+      new_password:
+        minimum_length: Salasanan pitää olla vähintään 2 merkkiä
+      new_password_repeat:
+        not_match: Salasanat eivät täsmää
```
Comment (pierallard) [inline src/Akeneo/UserManagement/Bundle/Resources/translations/messages.fi_FI.yml]: Sköl Offenstru

**Comment by pierallard** [comment]
Warning, there are "mass edit rules" translations in CE... Weird.

File: src/Akeneo/Tool/Bundle/VersioningBundle/Event/PrePurgeVersionEvent.php (hunk 3 above)
Comment (mmetayer) [reply src/Akeneo/Tool/Bundle/VersioningBundle/Event/PrePurgeVersionEvent.php]: To remove on master (can't do BC breaks anymore on 4.0...)

File: src/Akeneo/UserManagement/Bundle/Resources/translations/messages.fi_FI.yml (hunk 5 above)
Comment (mmetayer) [reply src/Akeneo/UserManagement/Bundle/Resources/translations/messages.fi_FI.yml]: it's Finnish, not Swedish :smile:

**Pull request merged**